### PR TITLE
Fix PostCaptures Accidentally Displayed on Capture tab.

### LINKED ISSUE #379: 升級到 0.12.1 以後，原本已上傳的照片出現驚嘆號，且 PostCapture 跑到 Capture 頁面
## Description
升級到 0.12.1 以後，原本已上傳的照片出現驚嘆號，且 PostCapture 跑到 Capture 頁面

## Steps to Reproduce
1. Go to Play store and update app
2. Open app

## Environment

- Version: 0.12.1
- OS:Android 10
- Device: Samsung 5G

### PR DESCRIPTION
**Pull request by seanwu1105**
Fix #379 and #383 by removing pre-fetching `Proof`s and disk cache of `DiaBackendAsset`s.

Previously, the app will assume ALL Captures and PostCaptures are stored (in the form of `Proof`) in local, and thus require pre-feching `Proof`s. With this fix, the app will first fetch the data from the backend then the local `Proof`s. By doing this, the issues #379 and #383 should be fixed as there would be no permanent local cache of `DiaBackendAsset`s. Everything would stay synced to the backend.

The implementation is a little bit messy regarding the part joining `DiaBackendAsset`s and `Proof`s. A better implementation could be found in [this experimental repo](https://github.com/numbersprotocol/capture-lite-performance/blob/develop/src/app/features/home/capture/capture-item/capture-item.component.ts#L35), which provides an uniform interface for `DiaBackendAsset`, `Proof` and `Assets`. However, the "better" implementation is still unstable. The messy implementation will be refactored in the future.

### CODE REVIEW

File: src/app/pages/home/home.page.ts [hunk 1]
```diff
@@ -100,37 +100,24 @@ export class HomePage implements OnInit {
...
-        Promise.all(
-          proofs.map(async proof => ({
-            proof,
-            thumbnailBase64: await proof.getThumbnailBase64(),
-          }))
-        )
-      )
-    );
-
     return combineLatest([
```
Comment (shc261392) [inline src/app/pages/home/home.page.ts:103]: I've found that this observable will not emit value if following the steps:

1. On browser, clean all previous data
2. Login with a clean account
3. Disable network and take 1 capture
4. Even if reload the page, it still won't emit

I did a bit of console log tracking and finds something weird. It's the `proofRepository.getAll$()` that didn't emit value. It seems in capacitor-filesystem-table, the `initialize()` never returns.

File: src/app/services/dia-backend/asset/dia-backend-asset-repository.service.ts [hunk 2]
```diff
@@ -1,87 +1,54 @@
...
   providedIn: 'root',
 })
 export class DiaBackendAssetRepository {
-  private readonly table = this.database.getTable<DiaBackendAsset>(
-    DiaBackendAssetRepository.name
-  );
   private readonly _isFetching$ = new BehaviorSubject(false);
-  private isDirty = true;
+
+  private readonly fetchAllCache$ = new BehaviorSubject<DiaBackendAsset[]>([]);
```
Comment (shc261392) [inline src/app/services/dia-backend/asset/dia-backend-asset-repository.service.ts:33]: By using a in-memory cache, does that mean it's not possible to determine if an asset has been uploaded or not if the user is offline?

For example, will the user see the `!` mark on all previous captures when trying to use Capture offline?

File: src/app/pages/home/home.page.ts (hunk 1 above)
Comment (shc261392) [reply src/app/pages/home/home.page.ts:103]: It seems this won't happen if I enable the network again. And if I disable the network, the problem reappears.

**Review by shc261392 — CHANGES_REQUESTED** [review]

File: src/app/pages/home/asset/asset.page.ts [hunk 3]
```diff
@@ -35,44 +39,17 @@ export class AssetPage {
...
   );
-  readonly location$ = this.capture$.pipe(
-    map(capture => [
-      capture.proofWithOld?.proof.geolocationLatitude,
-      capture.proofWithOld?.proof.geolocationLongitude,
-    ]),
-    map(([latitude, longitude]) => {
+  readonly location$ = this.asset$.pipe(
+    map(asset => {
+      const latitude = asset.information.information.find(
```
[the remark this answers is not in the document]
Comment (seanwu1105) [reply src/app/pages/home/asset/asset.page.ts:47]: @shc261392 The [documentation](https://dia-backend-dev.numbersprotocol.io/redoc/#operation/api_v2_assets_read) indicates `meta` is __required__. Where can I find this piece of information?

**Review by shc261392 — APPROVED** [review]

**Pull request merged**